hoomd/ParticleData.cc

```diff
@@ -199,9 +199,41 @@ ParticleData::~ParticleData()
 
 /*! \return Simulation box dimensions
  */
-const BoxDim& ParticleData::getBox() const
+const BoxDim ParticleData::getBox() const
     {
-    return m_box;
+    return *m_box;
+    }
+
+/*! \param box New box dimensions to set
+    \note ParticleData does NOT enforce any boundary conditions. When a new box is set,
+        it is the responsibility of the caller to ensure that all particles lie within
+        the new box.
+*/
+void ParticleData::setGlobalBox(const std::shared_ptr<BoxDim>& box)
+    {
+    assert(box->getPeriodic().x);
+    assert(box->getPeriodic().y);
+    assert(box->getPeriodic().z);
+    if (!box)
+        {
+        throw std::runtime_error("Expected non-null shared pointer to a box.");
+        }
+    m_global_box = std::make_shared<const BoxDim>(*box);
+
+#ifdef ENABLE_MPI
+    if (m_decomposition)
+        {
+        bcast(m_global_box, 0, m_exec_conf->getMPICommunicator());
```

Comment:
This calls HOOMD's `bcast` in `HOOMDMPI.h`. Does it broadcast the contents of the box? Or the shared pointer value? In either case, I'd be more confident in this code it it broadcast a raw BoxDim object instead.